tests/test_runner/test_discover_runner.py

```diff
@@ -644,6 +644,24 @@ def test_log_logger(self):
                     runner.log('log message', level)
                 self.assertEqual(cm.output, [expected])
 
+    def test_suite_result_with_failure(self):
+        cases = [
+            (1, 'FailureTestCase'),
+            (1, 'ErrorTestCase'),
+            (0, 'ExpectedFailureTestCase'),
+            (1, 'UnexpectedSuccessTestCase'),
+        ]
+        for expected_failures, testcase in cases:
+            with self.subTest(testcase=testcase):
+                runner = DiscoverRunner(verbosity=0)
```

Comment:
We don't need to re-instantiate a runner
```suggestion
        cases = [
            (1, 'FailureTestCase'),
            (1, 'ErrorTestCase'),
            (0, 'ExpectedFailureTestCase'),
            (1, 'UnexpectedSuccessTestCase'),
        ]
        runner = DiscoverRunner(verbosity=0)
        for expected_failures, testcase in cases:
            with self.subTest(testcase=testcase):
```